modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultOperationContext.java

```diff
@@ -59,9 +62,30 @@ public DefaultOperationContext(ConfigurationInstance<Object> configuration, Reso
      * {@inheritDoc}
      */
     @Override
-    public Object getParameter(String parameterName)
+    public <T> T getParameter(String parameterName)
     {
-        return parameters.get(parameterName);
+        return (T) parameters.get(parameterName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T> T getTypeSafeParameter(String parameterName, Class<? extends T> expectedType)
+    {
+        T value = getParameter(parameterName);
```

Comment:
I think this must be inside a try/catch for ClassCastException and remove the isInstance check below